src/app/components/multiple-autocomplete/multiple-autocomplete.component.ts

```diff
@@ -27,20 +33,33 @@ export class MultipleAutocompleteComponent {
 
   @Input() placeholder: string;
   @Input() width: string;
+  @Input() callback: () => void;
 
   @ViewChild('searchInput') searchInput: ElementRef<HTMLInputElement>;
   @ViewChild('auto') matAutocomplete: MatAutocomplete;
 
   constructor() {
-    this.filteredOptions = this.searchCtrl.valueChanges.pipe(
-      startWith(null),
-      map((val: string | null) =>
-        val ? this._filter(val) : this.allOptions.slice()
-      )
-    );
+    // Debounce is set to 30ms, since value changes frequently, which is imperceptible to the user.
+    this.filteredOptions = this.searchCtrl.valueChanges
+      .pipe(debounceTime(30), distinctUntilChanged())
+      .pipe(
+        startWith(null),
+        map((val: string | null) =>
+          val ? this._filter(val) : this.allOptions.slice()
+        )
+      );
   }
 
-  add(event: MatChipInputEvent): void {
+  /* Runs when the model changes (ngModelChange)
+    To throttle being called, the debounce is set to 60 ms.
+  */
+  onChange = () => {
+    this.searchCtrl.valueChanges
+      .pipe(debounceTime(60), distinctUntilChanged())
+      .subscribe(this.callback);
+  };
```

Comment:
This is cool! The only problem I notice with this is the tree is not updated when a value is removed. Since this is a subscription, maybe all it would take to fix is update the `searchCtrl` when `remove()` is called, but I'm unsure what effect that would have.